src/persist-client/src/lib.rs

```diff
@@ -1262,6 +1277,99 @@ mod tests {
         assert_eq!(read.expect_snapshot_and_fetch(5).await, all_ok(&data, 5));
     }
 
+    #[tokio::test]
+    async fn writer_heartbeat() {
+        mz_ore::test::init_logging();
+
+        let data = vec![
+            (("1".to_owned(), "one".to_owned()), 1, 1),
+            (("2".to_owned(), "two".to_owned()), 2, 1),
+            (("3".to_owned(), "three".to_owned()), 3, 1),
+        ];
+
+        let shard_id = ShardId::new();
+        let now = Arc::new(AtomicU64::new(0));
+        let now_clone = Arc::clone(&now);
+        let mut cache = new_test_client_cache();
+        cache.cfg.now = NowFn::from(move || now_clone.load(Ordering::Relaxed));
```

Comment:
nit: I'd just use Ordering::SeqCst so a reader doesn't have to reason about whether Relaxed is correct (and since the test isn't perf critical). here and all the `fetch_add`s